netdisco/discoverables/__init__.py

```diff
@@ -36,8 +36,15 @@ def __init__(self, netdis):
 
     def get_info(self):
         """Get most important info, by default the description location."""
-        return list(set(
-            self.info_from_entry(entry) for entry in self.get_entries()))
+        entries = self.get_entries()
+
+        # return list allowing for entries with both legacy tuples as well as
+        # later dictionaries
+        return list(
+            set(self.info_from_entry(entry)) if isinstance(entry, tuple)
```

Comment:
I was getting an 'unhashable' Python error, I think because of the use of 'set'. If we remove 'set' then likely it'll work fine with a tuple and a dict. I can check.